docs/rules/modeling.md

```diff
@@ -429,3 +429,27 @@ or any other nested information.
   ```
 
 ![A refactored DAG showing three sources which are each being referenced by an accompanying staging model](https://user-images.githubusercontent.com/30663534/159603703-6e94b00b-07d1-4f47-89df-8e5685d9fcf0.png){ width=500 }
+
+---
+
+## Models with Too Many Joins
+
+`fct_too_many_joins` ([source](https://github.com/dbt-labs/dbt-project-evaluator/tree/main/models/marts/dag/fct_too_many_joins.sql)) shows models with a reference to too many other models or sources.
+
+The number of different references to start raising errors is set to 7 by default, but you can set your own threshold by overriding the `too_many_joins_threshold` variable. [See overriding variables section.](../customization/overriding-variables.md)
+
+**Example**
+
+`fct_model_1` directly references seven (7) staging models upstream.
+
+![A DAG showing a model that directly references seven staging models upstream.](https://github.com/BradCr/dbt-project-evaluator/assets/151274228/46ea1f78-1bd7-436b-b15b-f63c726601a1){ width=600 }
+
+**Reason to Flag**
+
+This likely represents a model in which too much is being done. Having a model that too many upstream models introduces a lot of code complexity, which can be challenging to understand and maintain.
+
+**How to Remediate**
+
+Bringing together a reasonable number (typically 4 to 6) of entities or concepts (staging models, or perhaps other intermediate models) that will be joined with another similarly purposed intermediate model to generate a mart. Rather than having too many joins in the mart, we can join two intermediate models that each house a piece of the complexity, giving us increased readability, flexibility, testing surface area, and insight into our components.
```

Comment:
```suggestion
Bringing together a reasonable number (typically 4 to 6) of entities or concepts (staging models, or perhaps other intermediate models) that will be joined with another similarly purposed intermediate model to generate a mart. Rather than having too many joins, we can join two intermediate models that each house a piece of the complexity, giving us increased readability, flexibility, testing surface area, and insight into our components.
```